src/cpp/session/modules/quarto/SessionQuarto.cpp

```diff
@@ -148,6 +148,15 @@ std::tuple<FilePath,Version> userInstalledQuarto()
    return std::make_tuple(FilePath(), Version());
 }
 
+#ifndef QUARTO_ENABLED
+
+void detectQuartoInstallation()
```

Comment:
The only other bit to get right here is that the Quarto preference pane should not show on the client side. It currently checks the `quarto_enabled` user preference for "hidden". If its awkward to force the value of this pref on Arm (e.g because we might need it reset at some point when Arm support comes online) then you could just forward the "Arm" state to the client somehow and check that.

Reply:
There is a pref layer called `computed` that is intended for this purpose (prefs whose default values are environment specific. https://github.com/rstudio/rstudio/blob/main/src/cpp/session/prefs/UserPrefsComputedLayer.cpp